Sources/Commands/SwiftPackageRegistryTool.swift

```diff
@@ -0,0 +1,183 @@
+            }
+
+            if let scope = scope {
+                configuration.scopedRegistries[scope] = .init(url: url)
+            } else {
+                configuration.defaultRegistry = .init(url: url)
+            }
+
+            try configuration.writeToJSONFile(in: localFileSystem, at: path)
+
+            // TODO: Add login and password to .netrc
+        }
+    }
+
+    struct Unset: SwiftCommand {
+        static let configuration = CommandConfiguration(
+            abstract: "Remove a configured registry")
+
+        @OptionGroup(_hiddenFromHelp: true)
+        var swiftOptions: SwiftToolOptions
+
+        @Flag(help: "Apply settings to all projects for this user")
+        var global: Bool = false
+
+        @Option(help: "Associate the registry with a given scope")
+        var scope: String?
+
+        func run(_ swiftTool: SwiftTool) throws {
+            let path = try swiftTool.getRegistryConfigurationPath(fileSystem: localFileSystem, global: global)
+            var configuration = try RegistryConfiguration.readFromJSONFile(in: localFileSystem, at: path)
+
+            if let scope = scope {
+                guard let _ = configuration.scopedRegistries[scope] else {
+                    throw RegistryConfigurationError.missingScope(scope)
+                }
+                configuration.scopedRegistries.removeValue(forKey: scope)
+            } else {
+                guard let _ = configuration.defaultRegistry else {
+                    throw RegistryConfigurationError.missingScope()
+                }
+                configuration.defaultRegistry = nil
+            }
+
+            try configuration.writeToJSONFile(in: localFileSystem, at: path)
+        }
+    }
+}
+
+// MARK: -
+
+private extension Decodable {
+    static func readFromJSONFile(in fileSystem: FileSystem, at path: AbsolutePath) throws -> Self {
+        let content = try fileSystem.readFileContents(path)
+
+        let decoder = JSONDecoder()
+
+        return try decoder.decode(Self.self, from: Data(content.contents))
+    }
+}
+
+private extension Encodable {
+    func writeToJSONFile(in fileSystem: FileSystem, at path: AbsolutePath) throws {
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = [.sortedKeys, .prettyPrinted, .withoutEscapingSlashes]
+
+        let data = try encoder.encode(self)
+
+        try fileSystem.writeFileContents(path, bytes: ByteString(data), atomically: true)
+    }
+}
+
+private extension SwiftTool {
+    func getRegistryConfigurationPath(fileSystem: FileSystem = localFileSystem, global: Bool) throws -> AbsolutePath {
+        let filename = "registries.json"
+        if global {
+            return try fileSystem.getOrCreateSwiftPMConfigDirectory().appending(component: filename)
+        } else {
+            let directory = try configFilePath()
+            if !fileSystem.exists(directory) {
+                try fileSystem.createDirectory(directory, recursive: true)
+            }
```

Comment:
@tomerd I saw your changes to `getOrCreateSwiftPMConfigDirectory` from #3430, but wasn't sure about what that means for the `.swiftpm/config` directory in package root. Is that still a thing? If so, what's the preferred way to create that / migrate existing `.swiftpm/config` files?

Reply:
its a good question @mattt. we certainly do have people using the existing mirrors file, so we would need some migration story there. Its almost feels like we need a separate PR to refactor `Workspace.Configuration` and the handling of the mirror file so we can then make these two code paths work more seamlessly together. One can also argue we should allow for a global mirror file, as we are doing for registries config

WDYT?

Reply:
here is a draft PR for what I think this could look like: https://github.com/apple/swift-package-manager/pull/3652